### PR TITLE
Modify go test filter and upload test coverage artifacts in go.yml

### PR DESCRIPTION
**Pull request by msftcoderdjw**
**Changes**

1. Found go.yml now use directory filter when runinng API and COA tests. Removed it to increase test coverage.
2. Upload test coverage files to action artifacts for developers debugging.

### CODE REVIEW

**Review by Haishi2016 — COMMENTED** [review]
Should we start to display coverage badge on the home page as well?

**Comment by msftcoderdjw** [comment]
> Should we start to display coverage badge on the home page as well?

Great idea. I am trying to display coverage badge. But the github action failed to create `shields` branch and commit the shields.json for badge display. Can you please help to grant the permission? @Haishi2016 
image

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       run: |
         mkdir -p /home/runner/go
         echo "export GOPATH=/home/runner/go" >> $HOME/.bashrc
-        echo "export PATH=\$PATH:\$GOPATH/bin" >> $HOME/.bashrc
+        echo "export PATH=\$path:\$GOPATH/bin" >> $HOME/.bashrc
```
Comment (Haishi2016) [inline .github/workflows/go.yml:33]: Shouldn't this be $PATH (upper-case) ?

Comment (msftcoderdjw) [reply .github/workflows/go.yml]: You are right, fixed.

**Comment by msftcoderdjw** [comment]
Hi @Haishi2016 , I have dropped the badge changes, please help to review again. Thanks.

**Review by Haishi2016 — APPROVED** [review]

**Pull request merged**